src/libNetwork/RumorManager.cpp

```diff
@@ -491,6 +491,8 @@ void RumorManager::SendMessage(const Peer& toPeer,
   }
 
   // Send the message to peer .
+  std::this_thread::sleep_for(
```

Comment:
Maybe check if (SIMULATED_NETWORK_DELAY_IN_MS == 0) then do a sleep_for, otherwise sleep 0 will still do the thread context switch.